plugins/reporters/aosd/src/main/kotlin/AudiOpenSourceDiagnosticsReporter.kt

```diff
@@ -0,0 +1,147 @@
+                .map { it.toCoordinates() }
+
+            val dependencies = input.ortResult.getPackages(omitExcluded = true).map {
+                it.metadata.toExternalDependency(input)
+            }
+
+            runCatching {
+                val model = AOSD2(directDependencies = directDependencies, dependencies = dependencies)
+                val projectName = project.id.toPath("-")
+
+                outputDir.resolve("aosd.$projectName.json").writeReport(model)
+            }
+        }
+
+        return reportFiles
+    }
+}
+
+private fun Package.toExternalDependency(input: ReporterInput): ExternalDependency =
+    ExternalDependency(
+        id = id.toCoordinates(),
+        name = id.name,
+        scmUrl = vcsProcessed.url.takeUnless { it.isEmpty() },
+        description = description.takeUnless { it.isEmpty() },
+        version = id.version,
+        licenses = toLicenses(input),
+        deployPackage = binaryArtifact.toDeployPackage(),
+        externalDependencies = input.ortResult.getDependencies(id, maxLevel = 1, omitExcluded = true).map {
+            it.toCoordinates()
+        }
+    )
+
+private fun Package.toLicenses(input: ReporterInput): List<AOSD2.License> {
+    val licenses = mutableListOf<AOSD2.License>()
+    val resolvedLicenseInfo = input.licenseInfoResolver.resolveLicenseInfo(id).filterExcluded()
+
+    fun getLicenses(
+        licenseView: LicenseView,
+        origin: AOSD2.Origin,
+        copyrights: List<String> = emptyList()
+    ): List<AOSD2.License> {
+        val effectiveLicense = resolvedLicenseInfo.effectiveLicense(
+            licenseView,
+            input.ortResult.getPackageLicenseChoices(id),
+            input.ortResult.getRepositoryLicenseChoices()
+        )
+
+        return effectiveLicense?.decompose()?.map { licenseExpression ->
+            val name = licenseExpression.toString()
+            val text = input.licenseTextProvider.getLicenseText(name)
+
+            AOSD2.License(
+                name = name,
+                spdxId = SpdxLicense.forId(name)?.id,
+                text = text.orEmpty(),
+                copyrights = copyrights.takeUnless { it.isEmpty() }?.let { AOSD2.Copyrights(copyrights) },
+                origin = origin
+            )
+        }.orEmpty()
+    }
+
+    val copyrights = resolvedLicenseInfo.getCopyrights().toList()
+
+    // Declared licenses come from package management metadata.
+    licenses += getLicenses(LicenseView.ONLY_DECLARED, AOSD2.Origin.PACKAGE_MANAGEMENT)
+
+    // Group licenses detected by a scanner by their provenance / origin.
+    val povenance = input.ortResult.scanner?.provenances?.find { it.id == id }?.packageProvenance
+    when (povenance) {
+        is RepositoryProvenance -> licenses += getLicenses(
+            LicenseView.ONLY_DETECTED,
+            AOSD2.Origin.SCM,
+            copyrights
+        )
```

Comment:
Don't we format multi-line when cases always with curly braces?

Reply:
I guess we usually do... it was formatted that way by IntelliJ, and since Detekt did not complain, I left it that way to avoid some additional nested / lines. Do you prefer to have curly braces?

Reply:
No, actually I would prefer to omit curly braces in such cases.